### PR TITLE
Update all libraries to the latest stable version and remove jetifier

### PR DESCRIPTION
**Pull request by TimoPtr**
I did remove Jetifier since it was not needed anymore, this library was the last one using jetifier in our project and was causing some issues.

I took the opportunity to update all the libraries to the latest stable version available.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -18,6 +18,6 @@
 # org.gradle.parallel=true
 
 group=me.tatarka.bindingcollectionadapter2
-version=4.0.0
+version=4.0.1
 android.useAndroidX=true
-android.enableJetifier=true
\ No newline at end of file
+android.enableJetifier=false
```
Comment (evant) [inline gradle.properties:23]: We can just delete this line no?

**Comment by evant** [comment]
In general I do want to be conservative on version updates as to not force the user to have to update themselves. I don't want to hamstring newer projects though so it would be helpful to pin-point the exactly what was causing the issue.

> this library was the last one using jetifier in our project and was causing some issues.

I'm surprised that this is the case, my understanding is jetifier only runs while building the project, so should not affect downstream users.

File: app/build.gradle [hunk 2]
```diff
@@ -40,17 +40,17 @@ dependencies {
     implementation project(':bindingcollectionadapter-recyclerview')
     implementation project(':bindingcollectionadapter-paging')
     implementation project(':bindingcollectionadapter-viewpager2')
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
```
Comment (evant) [inline app/build.gradle:43]: 👍 updating deps on the sample app and test dependencies is fine.

File: gradle.properties (hunk 1 above)
Comment (TimoPtr) [reply gradle.properties:23]: yep

**Comment by evant** [comment]
@TimoPtr I made some changes to be a bit more conservative on the updates and fix the tests, does this still solve your issue? https://github.com/evant/binding-collection-adapter/pull/214

**Comment by TimoPtr** [comment]
@evant sorry I hadn't the time to answer you yesterday ... The issue was with BR class duplicate and R8 being strict about it. I found out that it was a cache issue ... Like most of the time, but removing jetifier is good thing :+1: it speed up a bit the build time ! And I'm totally ok with your change and I totally understand your point. I was not sure of what was you strategy there when I did it. So all good for me and thanks for your reactivity ! And you nice library

**Pull request closed without merging**